### PR TITLE
Checkpoint attribute

### LINKED ISSUE #95: File persister being overwritten by receiver options
### Configuration

- Eventhub configured with offset persistence using `persist.FilePersister` 
- Receiver(s) configured with `ReceiveFromTimestamp`
- Static configuration (the configuration does not change between runs)

### Expected Behavior

The offset saved in the `FilePersister` directory of the hub optionally supersedes the offset given as receiver option.

### Actual Behavior

On receiver creation the offset in the `FilePersister` directory is immediately overwritten with the one from the receiver option (e.g. https://github.com/Azure/azure-event-hubs-go/blob/master/receiver.go#L104). As a consequence events are received multiple times if the application is restarted.

I can understand if this is by design and cannot be changed, in that case I'll create a workaround for my specific implementation. Ideally a setting could be created to which offset has preference, either specific for the file persister or for all persisters. 

### Environment
- OS: `Fedora 29`
- Go version: `go version go1.11.5 linux/amd64`
- Version of Library: `head` (commit b14f5f2a2500e9ecdb3d3e3332e2c0fabd9da5b6)

### PR DESCRIPTION
**Pull request by R290**
### Fix or Enhancement?

Fixes #95 & #118

### Environment
- OS: Ubuntu 19.04
- Go version: go1.12.7 linux/amd64

### CODE REVIEW

**Comment by devigned** [comment]
@gavinfish give this a look when you have a sec.

@R290 thank you for the PR. Code looks great. If you have a minute, could you add an entry to the changelog.

After @gavinfish has a chance to review, we'll get this pulled in. Cheers!

File: receiver.go [hunk 1]
```diff
@@ -84,21 +85,25 @@ func ReceiveWithConsumerGroup(consumerGroup string) ReceiveOption {
 // ReceiveWithStartingOffset configures the receiver to start at a given position in the event stream
 func ReceiveWithStartingOffset(offset string) ReceiveOption {
 	return func(receiver *receiver) error {
-		return receiver.storeLastReceivedCheckpoint(persist.NewCheckpoint(offset, 0, time.Time{}))
+		receiver.checkpoint = persist.NewCheckpoint(offset, 0, time.Time{})
+		return nil
 	}
+
```
Comment (gavinfish) [inline receiver.go]: Please remove unnecessary empty line here.

File: receiver.go [hunk 2]
```diff
@@ -145,8 +150,20 @@ func (h *Hub) newReceiver(ctx context.Context, partitionID string, opts ...Recei
...
+	oldCheckpoint, err := receiver.getLastReceivedCheckpoint()
+	if err == nil {
+		receiver.checkpoint = oldCheckpoint
+	}
+
+	err = receiver.storeLastReceivedCheckpoint(receiver.checkpoint)
+
+	if err != nil {
+		return nil, err
+	}
```
Comment (metrue) [inline receiver.go]: Not sure code style of Azure repo, but prefer 
```
	if err = receiver.storeLastReceivedCheckpoint(receiver.checkpoint); err != nil {
		return nil, err
	}
```

File: receiver.go [hunk 3]
```diff
@@ -145,8 +150,20 @@ func (h *Hub) newReceiver(ctx context.Context, partitionID string, opts ...Recei
...
+
 	tab.For(ctx).Debug("creating a new receiver")
-	err := receiver.newSessionAndLink(ctx)
+	err = receiver.newSessionAndLink(ctx)
 	return receiver, err
```
Comment (metrue) [inline receiver.go]:
```
return receiver, receiver.newSessionAndLink(ctx)
```

File: receiver.go [hunk 4]
```diff
@@ -145,8 +150,20 @@ func (h *Hub) newReceiver(ctx context.Context, partitionID string, opts ...Recei
 		}
 	}
 
+	// update checkpoint if old checkpoint is succesfully read from e.g. file or memory
+	oldCheckpoint, err := receiver.getLastReceivedCheckpoint()
+	if err == nil {
```
[the remark this answers is not in the document]
Comment (R290) [reply receiver.go]: When `err` is not nil the `receiver.checkpoint` is not overwritten with `oldCheckpoint`. In that case the default checkpoint or the checkpoint set by the ` ReceiveOption` is used. In my opinion there is no need to do this explicitly.

**Pull request merged**